### PR TITLE
Refactor/#43-upgrade-build-system

### LINKED ISSUE #43: Upgrade build system
* Upgrade dependencies and plugins to latest versions
* Migrate from Travis CI to GitHub Actions

### PR DESCRIPTION
**Pull request by kaklakariada**
Closes #43

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,18 +1,17 @@
...
     <groupId>com.exasol</groupId>
     <artifactId>postgresql-virtual-schema</artifactId>
-    <version>2.0.0</version>
+    <version>2.0.1</version>
     <name>Virtual Schema for PostgreSQL</name>
     <properties>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
         <java.version>11</java.version>
         <surefire.and.failsafe.plugin.version>3.0.0-M5</surefire.and.failsafe.plugin.version>
```
Comment (jakobbraun) [inline pom.xml:12]: 3.0.0 was released

File: pom.xml [hunk 2]
```diff
@@ -139,13 +129,14 @@
...
-            <version>0.1.0</version>
+            <version>0.1.1</version>
             <scope>test</scope>
         </dependency>
         <dependency>
             <groupId>com.exasol</groupId>
             <artifactId>virtual-schema-shared-integration-tests</artifactId>
-            <version>1.0.0</version>
+            <version>2.1.0</version>
+            <scope>test</scope>
```
Comment (jakobbraun) [inline pom.xml:139]: :+1:

File: src/test/java/com/exasol/adapter/dialects/postgresql/PostgreSQLScalarFunctionsIT.java [hunk 3]
```diff
@@ -1,137 +1,126 @@
...
+                        for (final Column column : tableRequest.getColumns()) {
+                            tableBuilder.column(column.getName().toLowerCase(), column.getType());
+                        }
+                        final Table table = tableBuilder.build();
+                        for (final List<Object> row : tableRequest.getRows()) {
+                            table.insert(row.toArray());
+                        }
+                    }
+
+                    final VirtualSchema virtualSchema = SETUP.createVirtualSchema(postgresSchema.getName(), Map.of());
```
Comment (jakobbraun) [inline src/test/java/com/exasol/adapter/dialects/postgresql/PostgreSQLScalarFunctionsIT.java:49]:
```suggestion
                    final VirtualSchema virtualSchema = SETUP.createVirtualSchema(postgresSchema.getName(), Collections.emptyMap());
```

File: src/test/java/com/exasol/adapter/dialects/postgresql/PostgreSQLScalarFunctionsIT.java [hunk 4]
```diff
@@ -1,137 +1,126 @@
...
-    @Override
-    protected SingleRowSingleTableVirtualSchemaTestSetup<Boolean> createBooleanVirtualSchemaTable() {
-        return createTestTable("boolean");
-    }
+            @Override
+            public Set<String> getDialectSpecificExcludes() {
+                return Set.of(
+                        // expected was a value close to <1970-03-01> (tolerance: +/- <0.00010>) but was
+                        // "1970-03-01T00:00:00Z"
+                        "add_months",
```
Comment (jakobbraun) [inline src/test/java/com/exasol/adapter/dialects/postgresql/PostgreSQLScalarFunctionsIT.java:78]: did this happen with the latest version of hamcrest-result-set-matcher? Or did you exclude it before updating?

**Review by jakobbraun — CHANGES_REQUESTED** [review]

File: src/test/java/com/exasol/adapter/dialects/postgresql/PostgreSQLScalarFunctionsIT.java (hunk 4 above)
Comment (kaklakariada) [reply src/test/java/com/exasol/adapter/dialects/postgresql/PostgreSQLScalarFunctionsIT.java:78]: It happened with the latest version.

File: pom.xml (hunk 1 above)
Comment (kaklakariada) [reply pom.xml:12]: Sorry, versions-maven-plugin does not find a newer version

**Review by jakobbraun — APPROVED** [review]

File: pom.xml (hunk 1 above)
Comment (jakobbraun) [reply pom.xml:12]: ok, my fault

**Pull request merged**